tests.py

```diff
@@ -0,0 +1,147 @@
+
+# Color constant to console print
+HEADER = '\033[95m'
+OKBLUE = '\033[94m'
+OKGREEN = '\033[92m'
+WARNING = '\033[93m'
+ENDC = '\033[0m'
+
+
+class Test:
+    """
+    A little class to iterate one or some check
+    on the project
+    """
+
+    # List of test function we want to execute
+    COMMAND_LIST = list()
+
+    def __init__(self):
+        # List all test method
+        methods = inspect.getmembers(self)
+        for method in methods:
+            if method[0][:15] == '_Test__command_':
+                self.COMMAND_LIST.append(method[0][15:])
+
+    def help(self):
+        print('Usage: tests [command [command]]')
+        print('Example: tests test coverage')
+        print()
+        print('Commands:')
+        for command in self.COMMAND_LIST:
+            print(" |-" + command)
+
+    def launch_commands(self, list_name=COMMAND_LIST):
+        """
+        Method to launch a list of tests
+        """
+        is_broken = False
+
+        for command_name in list_name:
+            # Launch the test
+            self.launch_command(command_name)
+
+            # User input to stop or continue
+            if command_name != list_name[-1]:
+                next = input("Do you want to continue (Y/N) ? [Y]")
+                if next in ['Yes', 'yes', 'Y', 'y', '']:
+                    pass
+                else:
+                    is_broken = True
+                    break
+
+        if is_broken:
+            print(WARNING + "Tests canceled." + ENDC)
+        else:
+            print(OKGREEN + "OK - Tests successful." + ENDC)
+
+    def launch_command(self, command_name):
+        """
+        Method to launch a specific command
+        :param command_name: The name of the test method
+        :return: None
+        """
+        if command_name in self.COMMAND_LIST:
+            # Get the method and call it to know the test case
+            method = getattr(self, '_Test__command_' + command_name)
+            name, description, commands = method()
+
+            # Execute the command
+            self.__execute_command(name, description, commands)
+        else:
+            raise ValueError('`' + command_name + '` is not a valid command name.')
+
+    @staticmethod
+    def __execute_command(name, description, commands):
+        message = "Running -- " + name + "..."
+        separator = "-" * len(message)
+
+        print(OKGREEN + separator + ENDC)
+        print(OKGREEN + message + ENDC)
+        print(OKGREEN + separator + ENDC)
+        print(OKBLUE + description + ENDC)
+
+        for command in commands:
+            print(HEADER + command + ENDC)
+            os.system(command)
+
+            if command != commands[-1]:
+                input("Press enter to continue...")
+
+        print(OKBLUE + "Runned -- " + name + "." + ENDC)
+
+    @staticmethod
+    def __command_coverage():
+        """
+        Method to test coverage of all the application
+        """
+        name = "Tests and Coverage"
```

Comment:
Why are tests linked to coverage? Wouldn't it be better if we could execute them separately?

Commands:
- Test
- Coverage
- Style

Reply:
Yes sorry, i forgot to change my screenshot (here is the new version).

Actually i merged test and coverage because we need to run tests to have the coverage so it was really strange to separate them in practice as i made in the first time.

When i wanted to have just coverage (or coverage before test) it was really strange to have an error on testing.

Reply:
Aaaahhh I see! That makes sense :)